lib/service/service.go

```diff
@@ -1074,6 +1074,11 @@ func NewTeleport(cfg *Config, opts ...NewTeleportOption) (*TeleportProcess, erro
 
 	process.RegisterFunc("common.rotate", process.periodicSyncRotationState)
 
+	// run one upload completer per-process if we're using async recording
+	if !services.IsRecordSync(cfg.Auth.SessionRecordingConfig.GetMode()) {
```

Comment:
I am leaning towards removing this check and always running the uploader service.

The concern here is if you use dynamic configuration to change the recording mode while the process is running we'll need to then start up the uploader, which feels like a pain to coordinate.

Reply:
I think if we won't to shoot for the starts, ideally we would isolate all of the uploader logic to `initUploadService`, and always call it. Then, we can determine the recording strategy, and set the `StreamEmitter` a single time for all services. In the case of node sync, we simply wouldn't start the uploader service and would set the `StreamEmitter` to the `AuthClient`.

This would remove the need for the duplicated `NewStreamer` logic in every service, as well as the complicated, impossible to follow, wrapping of streamers with `AsyncStreamer`, `TeeStreamer`, etc. Not an easy task, but I would go all or nothing personally :D